app/services/dashtag/api_service.rb

```diff
@@ -61,8 +63,9 @@ def pull_instagram_posts_from_users_and_parse(user_id)
         GramParser.parse(response.parsed_response)
       end
 
-      def pull_twitter_posts_and_parse(hashtag)
-        response = HTTParty.get("https://api.twitter.com/1.1/search/tweets.json?q=%23#{hashtag}",
+      def pull_twitter_posts_and_parse(hashtags)
+        hashtag_query = hashtags.map { |hashtag| "%23#{hashtag}" }.join("%20AND%20")
+        response = HTTParty.get("https://api.twitter.com/1.1/search/tweets.json?q=#{hashtag_query}",
         :headers => { "Authorization" => "Bearer #{twitter_bearer_token}",
```

Comment:
Oh thank you good catch.  I'll take a look at it during lunch.

On Fri, May 29, 2015 at 10:47 AM, anirudh-eka <notifications@github.com>
wrote:

> In app/services/dashtag/api_service.rb
> <https://github.com/anirudh-eka/dashtag/pull/99#discussion_r31336772>:
>
> > @@ -61,8 +63,9 @@ def pull_instagram_posts_from_users_and_parse(user_id)
> >          GramParser.parse(response.parsed_response)
> >        end
> >
> > -      def pull_twitter_posts_and_parse(hashtag)
> > -        response = HTTParty.get("https://api.twitter.com/1.1/search/tweets.json?q=%23#{hashtag} <https://api.twitter.com/1.1/search/tweets.json?q=%23#%7Bhashtag%7D>",
> > +      def pull_twitter_posts_and_parse(hashtags)
> > +        hashtag_query = hashtags.map { |hashtag| "%23#{hashtag}" }.join("%20AND%20")
> > +        response = HTTParty.get("https://api.twitter.com/1.1/search/tweets.json?q=#{hashtag_query}",
> >          :headers => { "Authorization" => "Bearer #{twitter_bearer_token}",
>
> Hey just noticed that twitter_bearer_token variable isn't set anywhere...
> So this would be null
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anirudh-eka/dashtag/pull/99/files#r31336772>.
>



-- 
*Derek Gliwa*
Emaildgliwa@thoughtworks.com
Telephone+1 3098308614
[image: ThoughtWorks]
<http://www.thoughtworks.com/?utm_campaign=derek-gliwa-signature&utm_medium=email&utm_source=thoughtworks-email-signature-generator>